mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/jobcluster/JobClusterActor.java

```diff
@@ -1680,7 +1681,17 @@ public void onWorkerEvent(WorkerEvent r) {
             if(!JobHelper.isTerminalWorkerEvent(r)) {
                 logger.warn("Event from worker {} has no valid running job. Terminating worker ", r.getWorkerId());
                 Optional<String> host = JobHelper.getWorkerHostFromWorkerEvent(r);
-                mantisScheduler.unscheduleAndTerminateWorker(r.getWorkerId(), host);
+                Optional<IMantisJobMetadata> completedJobOptional =
+                    jobManager.getJobDataForCompletedJob(r.getWorkerId().getJobId());
+                if (completedJobOptional.isPresent()) {
+                    JobDefinition jobDefinition =
```

Comment:
nit: log

Reply:
Also shall we consider saving this state here into 'jobManager' in case some in-transit node keeps triggering this to create a new scheduler and attempt to unscheduled?

Reply:
Yea, I wanted to take a stab at this. But, unfortunately, that class is super dense - so I thought I'll clean it up later.